twilio-audio-router/src/main/java/com/twilio/audio_manager/AudioDevice.java

```diff
@@ -0,0 +1,32 @@
+package com.twilio.audio_manager;
+
+import androidx.annotation.NonNull;
+
+public class AudioDevice {
+    public final @NonNull String name;
+    public final @NonNull Type type;
+
+    public AudioDevice(@NonNull Type type, @NonNull String name) {
+        this.type = type;
+        this.name = name;
+    }
+
+    /** Audio device types */
+    public enum Type {
+        SPEAKERPHONE,
+        WIRED_HEADSET,
+        EARPIECE,
+        BLUETOOTH
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Nit: Override `hashCode` in addition to equals

Reply:
Good catch! Also, once we convert to Kotlin, we can use a data class here which automatically generates equals and hashCode functions 😄

Reply:
I'll add that hashcode method in the meantime though.